cors.go

```diff
@@ -47,21 +50,25 @@ func (c *Config) HandleRequest(w http.ResponseWriter, r *http.Request) {
 	if requestOrigin == "" {
 		return
 	}
+
 	//check origin against allowed origins
 	for _, ao := range c.AllowedOrigins {
 		if ao == "*" || ao == requestOrigin {
 			responseOrigin := "*"
 			if ao != "*" {
 				responseOrigin = requestOrigin
 			}
-			w.Header().Set(allowOriginKey, responseOrigin)
-			w.Header().Add(varyKey, originKey)
+			addAllowOriginHeader(w, responseOrigin)
 			break
 		}
 	}
 
 	if w.Header().Get(allowOriginKey) == "" {
-		return //if we didn't set a valid allow-origin, none of the other headers matter
+		if c.anyOriginRegexpMatch(requestOrigin) {
```

Comment:
Doing this here force users to also set an `origin`, as per default `w.Header().Get(allowOriginKey)` will return `"*"`.

What do you think of moving the rexexp check before `c.AllowedOrigins` check?

It's backward compatible as current users will get an empty slice that won't match, and allows to use  `origin_regexp` alone.

Reply:
I fixed this in b668583a86db3d94711ffae78d67da0e8ea1d528.